src/modules/dataset.ts

```diff
@@ -14,12 +15,22 @@ function updateDataset(oldVnode: VNode, vnode: VNode): void {
 
   for (key in oldDataset) {
     if (!dataset[key]) {
-      delete elm.dataset[key];
+      if (elm.dataset) {
+        delete elm.dataset[key];
+      } else {
+        key = key.replace(CAPS_REGEX, '-$&').toLowerCase();
+        elm.removeAttribute('data-' + key);
+      }
     }
   }
   for (key in dataset) {
     if (oldDataset[key] !== dataset[key]) {
-      elm.dataset[key] = dataset[key];
+      if (elm.dataset) {
+        elm.dataset[key] = dataset[key];
+      } else {
+        key = key.replace(CAPS_REGEX, '-$&').toLowerCase();
+        elm.setAttribute('data-' + key, dataset[key]);
```

Comment:
This is incorrect, the transformed `key` should only we used as the attribute's key, you can't use it for `dataset[key]`. I think we need tests for this to make sure we are doing the right thing.